bundles/org.openhab.binding.rotel/src/main/java/org/openhab/binding/rotel/internal/RotelModel.java

```diff
@@ -320,57 +333,40 @@ public boolean hasOtherThanPrimaryCommands() {
     }
 
     /**
-     * Inform whether zone 2 commands are available
+     * Inform whether zone N commands are available
      *
-     * @return true if zone 2 commands are available
-     */
-    public boolean hasZone2Commands() {
-        return nbAdditionalZones >= 1 && additionalCommands;
-    }
-
-    /**
-     * Inform whether zone 3 commands are available
+     * @param numZone the zone number, 1 for for zone 1 until 4 for zone 4
      *
-     * @return true if zone 3 commands are available
+     * @return true if zone N commands are available
      */
-    public boolean hasZone3Commands() {
-        return nbAdditionalZones >= 2 && additionalCommands;
+    public boolean hasZoneCommands(int numZone) {
+        if (numZone < 1 || numZone > 4) {
```

Comment:
Sorry, found a new constant you introduced which could be used for parameter validation:
```suggestion
        if (numZone < 1 || numZone > RotelBindingConstants.MAX_NUMBER_OF_ZONES) {
```

Reply:
Done